tensornetwork/backends/symmetric/symmetric_backend.py

```diff
@@ -163,7 +165,127 @@ def conj(self, tensor: Tensor) -> Tensor:
+    of a linear operator `A`. 
+    If no `initial_state` is provided then `shape`  and `dtype` are required
+    so that a suitable initial state can be randomly generated.
+    This is a wrapper for scipy.sparse.linalg.eigs which only supports
+    a subset of the arguments of scipy.sparse.linalg.eigs.
+    Note: read notes for `enable_caching` carefully.
+
+    Args:
+      A: A (sparse) implementation of a linear operator
+      args: A list of arguments to `A`.  `A` will be called as
+        `res = A(initial_state, *args)`.
+      initial_state: An initial vector for the algorithm. If `None`,
+        a random initial `Tensor` is created using the `numpy.random.randn`
+        method.
+      shape: The shape of the input-dimension of `A`.
+      dtype: The dtype of the input `A`. If both no `initial_state` is provided,
+        a random initial state with shape `shape` and dtype `dtype` is created.
+      num_krylov_vecs: The number of iterations (number of krylov vectors).
+      numeig: The nummber of eigenvector-eigenvalue pairs to be computed.
+        If `numeig > 1`, `reorthogonalize` has to be `True`.
+      tol: The desired precision of the eigenvalus. Uses
+      which : ['LM' | 'SM' | 'LR' | 'SR' | 'LI']
+        Which `k` eigenvectors and eigenvalues to find:
+            'LM' : largest magnitude
+            'SM' : smallest magnitude
+            'LR' : largest real part
+            'SR' : smallest real part
+            'LI' : largest imaginary part
+      maxiter: The maximum number of iterations.
+      enable_caching: If `True`, block-data during calls to `matvec` is cached
+        for later reuse. Note: usually it is save to enable_caching, unless 
+        `matvec` uses matrix decompositions like SVD, QR, eigh, eig or similar.
+        In this case, if one does a large number of krylov steps, this can lead 
+        to memory clutter and/or overflow.
+
+    Returns:
+       `np.ndarray`: An array of `numeig` lowest eigenvalues
+       `list`: A list of `numeig` lowest eigenvectors
+    """
+
+    if args is None:
+      args = []
+
+    if which in ('SI', 'LI'):
+      raise ValueError(f'which = {which} is currently not supported.')
+
+    if numeig + 1 >= num_krylov_vecs:
+      raise ValueError('`num_krylov_vecs` > `numeig + 1` required!')
+
+    if initial_state is None:
+      if (shape is None) or (dtype is None):
+        raise ValueError("if no `initial_state` is passed, then `shape` and"
+                         "`dtype` have to be provided")
+      initial_state = self.randn(shape, dtype)
+
+    if not isinstance(initial_state, BlockSparseTensor):
+      raise TypeError("Expected a `BlockSparseTensor`. Got {}".format(
+          type(initial_state)))
+    
+    former_caching_status = self.bs.get_caching_status()
+    self.bs.set_caching_status(enable_caching)
+    if enable_caching:
+      cache_was_empty = self.bs.get_cacher().is_empty
+    try:
```

Comment:
There's a lot of code inside this try/except - if possible move code which will not throw the exception out of this scope

Reply:
Anything that can potentially raise should be inside the try. The reason is that if an exception is raised, the routine will be left without resetting the caching status to it's prior value.

Reply:
I moved some code out of the try statement.